File: src/main/java/com/sslproxy/coordinator/util/Sha256Utils.java
package com.sslproxy.coordinator.util;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.util.HexFormat;

/**
 * SHA-256 hashing utilities matching the Zig sha256Hex function.
 */
public final class Sha256Utils {

    private Sha256Utils() {}

    /**
     * Computes SHA-256 hex digest.
     */
    public static String sha256Hex(byte[] payload) {
        try {
            MessageDigest digest = MessageDigest.getInstance("SHA-256");
            byte[] hash = digest.digest(payload);
            return HexFormat.of().formatHex(hash);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException("SHA-256 not available", e);
        }
    }

    /**
     * Computes SHA-256 hex digest of a UTF-8 string.
     */
    public static String sha256Hex(String payload) {
        return sha256Hex(payload.getBytes(java.nio.charset.StandardCharsets.UTF_8));
    }
}

File: src/main/scala/com/sslproxy/coordinator/util/Sha256Utils.scala
package com.sslproxy.coordinator.util

import java.nio.charset.StandardCharsets
import java.security.MessageDigest

object Sha256Utils:
  def sha256Hex(input: String): String =
    val digest = MessageDigest.getInstance("SHA-256")
    digest.update(input.getBytes(StandardCharsets.UTF_8))
    digest.digest().map("%02x".format(_)).mkString

  def sha256Hex(bytes: Array[Byte]): String =
    val digest = MessageDigest.getInstance("SHA-256")
    digest.update(bytes)
    digest.digest().map("%02x".format(_)).mkString
